python/src/main/resources/python/zeppelin_python.py

```diff
@@ -142,19 +152,33 @@ def handler_stop_signals(sig, frame):
           mod = ast.Module([node])
           code = compile(mod, '<stdin>', 'exec')
           exec(code, _zcUserQueryNameSpace)
+
+        if not isForCompletion:
+          # only call it when it is not for code completion. code completion will call it in
+          # PythonCompletion.getCompletion
+          intp.setStatementsFinished("", False)
+      except Py4JJavaError:
+        # raise it to outside try except
+        raise
       except:
-        raise Exception(traceback.format_exc())
+        if not isForCompletion:
+          exception = traceback.format_exc()
```

Comment:
add a comment what this is looking for and what it looks like typically?

Reply:
Fixed